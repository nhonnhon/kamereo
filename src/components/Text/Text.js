import React from 'react'
import PropTypes from 'prop-types'

import './Text.scss'

const Text = props => {
  const { value, className, children } = props
  const classNames = `text ${className || ''}`
  return (
    <p {...props} className={classNames}>
      {children || value}
    </p>
  )
}

Text.propTypes = {
  value: PropTypes.string,
  children: PropTypes.node,
  className: PropTypes.string,
}

export default Text
