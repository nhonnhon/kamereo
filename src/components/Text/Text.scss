@import './../../assets/css/base.scss';

.text {
  display: block;
  color: $black;
  text-align: left;

  &.uppercase {
    text-transform: uppercase;
  }

  &.inline {
    display: inline-block;

    &.top {
      vertical-align: top;
    }

    &.center {
      vertical-align: middle;
    }

    &.bot {
      vertical-align: bottom;
    }
  }

  &.clickable {
    &:hover {
      cursor: pointer;
    }
  }

  &.white {
    color: $white;
  }
}
