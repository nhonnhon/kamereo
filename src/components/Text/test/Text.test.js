import React from 'react'
import { create } from 'react-test-renderer'
import Text from '../Text'

describe('Text component', () => {
  test('Matches the snapshot', () => {
    const text = create(<Text />)
    expect(text.toJSON()).toMatchSnapshot()
  })
})
