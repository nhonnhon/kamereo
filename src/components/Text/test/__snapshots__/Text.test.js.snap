// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component Matches the snapshot 1`] = `
<p
  className="text "
/>
`;
